gstreamer-source.c

```diff
@@ -82,7 +134,6 @@ static gboolean bus_callback(GstBus *bus, GstMessage *message,
 		g_error_free(err);
 	} // fallthrough
 	case GST_MESSAGE_EOS:
-		gst_element_set_state(data->pipe, GST_STATE_NULL);
```

Comment:
I hope this is safe? Just wondering if something may continue to do something bad here. Probably not, but I have not tested :)

Reply:
I've tested this pretty extensively, and I'm reasonably certain that it's not required (or even useful really) to set GST_STATE_NULL "immediately" for an error condition here -- and it is very undesirable for the EOS condition.   It's fine to leave the pipeline in ERROR (forever or until it we destroy and recreate it).   It's very userful to leave the pipeline in the EOS state, because we can still seek back and resume playing -- works great in my testing with the OBS slider control (and watching with OBS-websocket to see the transition from "playing" to "eos" and then from "eos" back to "buffering->playing" when I rewind-seek).

I will go so far as to say that I believe we are required to set GST_STATE_NULL immediately before unref'ing data->pipe *and never at any other time than that*.

My [v4] implements such.